src/EnvironmentChecker.php

```diff
@@ -103,27 +104,42 @@ public function __construct($checkSuiteName, $title)
      */
     public function init($permission = 'ADMIN')
     {
+        $canAccess = $this->canAccess(null, $permission);
+        // Allow bypassing basic-auth check if the user is logged in with the required permission
+        if ($canAccess) {
+            return;
+        }
         // if the environment supports it, provide a basic auth challenge and see if it matches configured credentials
         if (Environment::getEnv('ENVCHECK_BASICAUTH_USERNAME')
             && Environment::getEnv('ENVCHECK_BASICAUTH_PASSWORD')
         ) {
-            // Check that details are both provided, and match
-            if (empty($_SERVER['PHP_AUTH_USER']) || empty($_SERVER['PHP_AUTH_PW'])
-                || $_SERVER['PHP_AUTH_USER'] != Environment::getEnv('ENVCHECK_BASICAUTH_USERNAME')
-                || $_SERVER['PHP_AUTH_PW'] != Environment::getEnv('ENVCHECK_BASICAUTH_PASSWORD')
+            $request = $this->getRequest();
+            $user = $request->getHeader('PHP_AUTH_USER');
+            $pw = $request->getHeader('PHP_AUTH_PW');
+            // Check that details match
+            if (empty($user) || empty($pw)
+                || $user !== Environment::getEnv('ENVCHECK_BASICAUTH_USERNAME')
+                || $pw !== Environment::getEnv('ENVCHECK_BASICAUTH_PASSWORD')
             ) {
                 // Fail check with basic auth challenge
                 $response = new HTTPResponse(null, 401);
                 $response->addHeader('WWW-Authenticate', "Basic realm=\"Environment check\"");
-                throw new HTTPResponse_Exception($response);
+                if ($user) {
+                    $response->setBody(_t(
```

Comment:
This is copied from https://github.com/silverstripe/silverstripe-framework/blob/5/src/Security/BasicAuth.php#L125